treeherder/webapp/api/logslice.py

```diff
@@ -58,9 +59,14 @@ def list(self, request, project, jm):
         if start_line >= end_line:
             return Response("``end_line`` must be larger than ``start_line``", 400)
 
-        if len(log) > 0:
+        # get only the log that matches the ``log_name``
+        logs = jm.get_log_references(job_id)
+        log = [log for log in logs if log["name"] == log_name]
+
+        if len(log):
+            log = log[0]
```

Comment:
Yet another great point!  That second one is the more "pythonic"  I like it.  Thanks for the tip!  I hadn't used ``next()`` before, tbh.